hal_st/stm32fxxx/DigitalToAnalogPinStm.hpp

```diff
@@ -15,14 +19,37 @@ namespace hal
         : public DigitalToAnalogPinImplBase
     {
     public:
+#if defined(DAC_CHIPCONNECT_BOTH)
+        // clang-format off
+        static const struct Internal{} internal;
+        static const struct External{} external;
+        static const struct Both{} both;
+
+        // clang-format on
+
+        enum struct Channel : uint32_t
+        {
+            channel_1 = 1,
+            channel_2 = 2,
+        };
+
+        DigitalToAnalogPinImplStm(Channel channel, hal::DacStm& dac, Internal);
+        DigitalToAnalogPinImplStm(hal::GpioPinStm& pin, hal::DacStm& dac, External);
+        DigitalToAnalogPinImplStm(hal::GpioPinStm& pin, hal::DacStm& dac, Both);
+#else
         DigitalToAnalogPinImplStm(hal::GpioPinStm& pin, hal::DacStm& dac);
+#endif
 
         void Set(uint16_t value) override;
 
     private:
-        AnalogPinStm pin;
+        infra::Optional<AnalogPinStm> pin;
         DacStm& dac;
         uint32_t channel;
+
+#if defined(DAC_CHIPCONNECT_BOTH)
+        const bool internallyConnected{ false };
```

Comment:
Since in some constructors internallyConnected is initialized to true, I would remove this default initialization and move it to the proper constructor